biorxiv/article_distances/05_biorxiv_article_distances_cosine.py

```diff
@@ -0,0 +1,440 @@
+)
+final_cosine_df.head()
+
+# # Distribution plot
+
+g = (
+    p9.ggplot(
+        final_original_df.replace(
+            {
+                "pre_vs_published": "preprint-published",
+                "pre_vs_random": "preprint-random",
+            }
+        )
+    )
+    + p9.aes(x="label", y="distance")
+    + p9.geom_violin(fill="#a6cee3")
+    + p9.labs(x="Document Pair Groups", y="Euclidean Distance")
+    + p9.theme_seaborn(context="paper", style="ticks", font="Arial", font_scale=2)
+    + p9.theme(figure_size=(11, 8.5))
+)
+print(g)
+
+g = (
+    p9.ggplot(
+        final_cosine_proxy_df.replace(
+            {
+                "pre_vs_published": "preprint-published",
+                "pre_vs_random": "preprint-random",
+            }
+        )
+    )
+    + p9.aes(x="label", y="distance")
+    + p9.geom_violin(fill="#a6cee3")
+    + p9.labs(x="Document Pair Groups", y="Euclidean (L2 Norm) Distance")
+    + p9.theme_seaborn(context="paper", style="ticks", font="Arial", font_scale=2)
+    + p9.theme(figure_size=(11, 8.5))
+)
+print(g)
+
+g = (
+    p9.ggplot(
+        final_cosine_df.replace(
+            {
+                "pre_vs_published": "preprint-published",
+                "pre_vs_random": "preprint-random",
+            }
+        )
+    )
+    + p9.aes(x="label", y="distance")
+    + p9.geom_violin(fill="#a6cee3")
+    + p9.labs(x="Document Pair Groups", y="Cosine Distance")
+    + p9.theme_seaborn(context="paper", style="ticks", font="Arial", font_scale=2)
+    + p9.theme(figure_size=(11, 8.5))
+)
+print(g)
+
+# # Recall at K
+
+data_rows = []
+for df, distance_label in zip(
+    [final_original_df, final_cosine_proxy_df, final_cosine_df],
+    ["euclidean", "euclidean (L2)", "cosine"],
+):
+    for k in tqdm.tqdm(range(1, 34503, 200)):
+        recall = recall_score(
+            df.sort_values("distance").iloc[0:k].label.tolist(),
+            ["pre_vs_published"] * k,
+            pos_label="pre_vs_published",
+        )
+
+        precision = precision_score(
+            df.sort_values("distance").iloc[0:k].label.tolist(),
+            ["pre_vs_published"] * k,
+            pos_label="pre_vs_published",
+        )
+
+        data_rows.append(
+            {
+                "recall": recall,
+                "precision": precision,
+                "N": k,
+                "distance": distance_label,
+            }
+        )
+
+plot_df = pd.DataFrame.from_records(data_rows)
+plot_df.head()
+
+g = (
+    p9.ggplot(plot_df, p9.aes(x="N", y="recall", color="distance"))
+    + p9.geom_point()
+    + p9.labs(x="Top N predictions", y="Recall")
+)
+g.save("output/figures/distance_metrics_top_k_recall.png", dpi=600)
+print(g)
+
+g = (
+    p9.ggplot(plot_df, p9.aes(x="N", y="precision", color="distance"))
+    + p9.geom_point()
+    + p9.labs(x="Top N predictions", y="Precision")
+)
+g.save("output/figures/distance_metrics_top_k_precision.png", dpi=600)
+print(g)
+
+# Take Home Points:
```

Comment:
SInce the distance metric doesn't seem to change your overall message, is the plan to use cosine distance since this is more standard? Also is there a reason you chose to use Euclidean in the first place, I forget

Reply:
>Since the distance metric doesn't seem to change your overall message, is the plan to use cosine distance since this is more standard? Also is there a reason you chose to use Euclidean in the first place, I forget

The opposite actually. The plan is to use these results to show that using a pure euclidean distance approach doesn't significantly affect performance. The main reason for Euclidean space was for runtime efficiency as Euclidean space runs faster than cosine distance (Euclidean distance is a true distance metric which is required by some algorithms).

Reply:
That makes sense. Thank you for clarifying